contracts/src/oracle/library.cairo

```diff
@@ -248,6 +259,27 @@ namespace Oracle {
     // Setters
     //
 
+    func publish_entry{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+        new_entry: GenericEntry
+    ) {
+        alloc_locals;
+
+        let (new_entry_ptr: GenericEntry*) = alloc();
+        assert new_entry_ptr[0] = new_entry;
+        validate_sender_for_source(cast(new_entry_ptr, felt*));
+
+        // TODO: Should we validate for key?  any publisher could publish this key
```

Comment:
what is the validation logic for a publisher publishing a key?

Reply:
We don't do this for future or spot entries, and I don't see a need to do this for generic entries right now. But agree this may be useful in the future (a few different ways to do this and open question how the tradeoffs are best resolved, permissioned publishing vs open publishing & curated feeds).